search/intermediate_result.cpp

```diff
@@ -116,192 +152,47 @@ PreResult2::PreResult2(FeatureType const & f, m2::PointD const & center, m2::Poi
-    {
-      coder->GetNearbyAddress(GetCenter(), addr);
-      addrComputed = true;
-    }
-    if (addr.GetDistance() == 0)
-      name = FormatStreetAndHouse(addr);
-  }
-
-  uint32_t const type = GetBestType(pTypes);
-
-  // Format full address only for suitable results.
-  string address;
-  if (coder)
-  {
-    address = GetRegionName(infoGetter, type);
-    if (ftypes::IsAddressObjectChecker::Instance()(m_types))
-    {
-      if (!addrComputed)
-      {
-        coder->GetNearbyAddress(GetCenter(), addr);
-        addrComputed = true;
-      }
-      address = FormatFullAddress(addr, address);
-    }
-  }
-
-  switch (m_resultType)
-  {
-  case RESULT_FEATURE:
-  case RESULT_BUILDING:
-    return Result(m_id, GetCenter(), name, address, pCat->GetReadableFeatureType(type, locale),
-                  type, m_metadata);
-  default:
-    ASSERT_EQUAL(m_resultType, RESULT_LATLON, ());
-    return Result(GetCenter(), name, address);
-  }
-}
-
-PreResult2::StrictEqualF::StrictEqualF(PreResult2 const & r, double const epsMeters)
-  : m_r(r), m_epsMeters(epsMeters)
-{
-}
-
-bool PreResult2::StrictEqualF::operator()(PreResult2 const & r) const
-{
-  if (m_r.m_resultType == r.m_resultType && m_r.m_resultType == RESULT_FEATURE)
-  {
-    if (m_r.IsEqualCommon(r))
-      return PointDistance(m_r.GetCenter(), r.GetCenter()) < m_epsMeters;
-  }
-
-  return false;
-}
-
-bool PreResult2::LessLinearTypesF::operator() (PreResult2 const & r1, PreResult2 const & r2) const
-{
-  if (r1.m_geomType != r2.m_geomType)
-    return (r1.m_geomType < r2.m_geomType);
-
-  if (r1.m_str != r2.m_str)
-    return (r1.m_str < r2.m_str);
-
-  uint32_t const t1 = r1.GetBestType();
-  uint32_t const t2 = r2.GetBestType();
-  if (t1 != t2)
-    return (t1 < t2);
-
-  // Should stay the best feature, after unique, so add this criteria:
-  return r1.m_distance < r2.m_distance;
-}
-
-bool PreResult2::EqualLinearTypesF::operator() (PreResult2 const & r1, PreResult2 const & r2) const
-{
-  // Note! Do compare for distance when filtering linear objects.
-  // Otherwise we will skip the results for different parts of the map.
-  return r1.m_geomType == feature::GEOM_LINE && r1.IsEqualCommon(r2) &&
-         PointDistance(r1.GetCenter(), r2.GetCenter()) < kDistSameStreetMeters;
-}
-
-bool PreResult2::IsEqualCommon(PreResult2 const & r) const
+bool RankerResult::IsEqualCommon(RankerResult const & r) const
 {
   return m_geomType == r.m_geomType && GetBestType() == r.GetBestType() && m_str == r.m_str;
 }
 
-bool PreResult2::IsStreet() const
+bool RankerResult::IsStreet() const
 {
   return m_geomType == feature::GEOM_LINE && ftypes::IsStreetChecker::Instance()(m_types);
 }
 
-string PreResult2::DebugPrint() const
+string RankerResult::DebugPrint() const
 {
   stringstream ss;
   ss << "IntermediateResult [ "
      << "Name: " << m_str
      << "; Type: " << GetBestType()
-     << "; Ranking info: " << search::DebugPrint(m_info)
+     << "; " << search::DebugPrint(m_info)
```

Comment:
Do we need ```search::``` here?

Reply:
We did, because it was confused with the method RankerResult::DebugPrint(), which I've now removed.
done